packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,192 @@
+          isError={isError}
+          label={label}
+          maxLength={maxLengthForLatestChainSpec}
+          onChange={this.onChange}
+          onKeyDown={this.onKeyDown}
+          onKeyUp={this.onKeyUp}
+          placeholder={t('account.balance.placeholder', {
+            defaultValue: 'Between 1 testnet DOT and the available testnet DOT balance (minus 1) of the account'
+          })}
+          withLabel={withLabel}
+        />
+      </Bare>
+    );
+  }
+
+  onChange = (value: string): void => {
+    const { onChange, t } = this.props;
+
+    // prepare value by remove all whitespace
+    value = value.split(' ').join('');
+
+    try {
+      const { isValid, errorMessage, infoMessage, inputConvertedFromScientificNotation } = isValidBalance(value);
+
+      this.setState({
+        error: !isValid && errorMessage ? t(errorMessage) : '',
+        info: isValid && infoMessage ? t(infoMessage) : ''
+      });
+
+      if (!onChange) return;
+
+      let newValue: BN;
+      if (inputConvertedFromScientificNotation) {
+        newValue = new BN(inputConvertedFromScientificNotation);
+      } else {
+        newValue = new BN(value || '0');
+      }
+
+      onChange({
+        isValid,
+        value: newValue
+      });
+    } catch (e) {
+      console.error('error: ', e);
+    }
+  }
+
+  onKeyDown = (event: any): void => {
+    const { t } = this.props;
+
+    const altKey = 18;
+    const arrowLeftKey = 37;
+    const arrowRightKey = 39;
+    const backspaceKey = 46;
+    const cmdKey = event.metaKey; // firefox (91), safari (224)
+    const ctrlKey = 17;
+    const decimalPointKey = 190; // decimals allowed for scientific or exponential notation
+    const delKey = 8;
+    const escapeKey = 27;
+    const enterKey = 13;
+    const plusKey = 187; // '+'
+    const tabKey = 9; // next input field
+    const aKey = 65; // select all from balance input
+    const xKey = 88; // cut balance
+    const cKey = 67; // copy balance
+    const eKey = 69; // scientific or exponential notation
+    const vKey = 86; // paste balance
+    const zeroKey = 48;
+    const zeroNumpadKey = 96;
+
+    const regexE = /[e]/gi;
+    const regexPlus = /[\+]/gi;
+    const regexDecimalPoint = /[\.]/gi;
+
+    // only allow user balance input to contain one instance of 'e', '+', and '.' for decimal in scientific or exponential notation
+    if (
+      (event.keyCode === eKey && event.target.value.match(regexE)) ||
+      (event.keyCode === plusKey && event.target.value.match(regexPlus)) ||
+      (event.keyCode === decimalPointKey && event.target.value.match(regexDecimalPoint))
+    ) {
+      event.preventDefault();
+      return;
+    }
+
+    // obtain the current cursor index position in the Balance input field
+    const inputCursorIndex = event.target.value.slice(0, event.target.selectionStart).length;
+
+    // allow these keys
+    if ([escapeKey, enterKey, backspaceKey, delKey, tabKey, ctrlKey, altKey, cmdKey, eKey, arrowLeftKey,
```

Comment:
`KEYS_ALLOWED` (Once again, we don't need to re-create the array each time, it is file constant)